google-cloud-contrib/google-cloud-nio/src/main/java/com/google/cloud/storage/contrib/nio/CloudStorageReadChannel.java

```diff
@@ -135,9 +135,33 @@ public int read(ByteBuffer dst) throws IOException {
     }
   }
 
+  private static boolean isReopenable(Throwable exs) {
+    Throwable throwable = exs;
+    while (throwable != null) {
+      if (throwable.getMessage().contains("Connection closed prematurely")
+          || throwable.getCause() instanceof javax.net.ssl.SSLException
+          || throwable.getCause() instanceof java.io.EOFException
+          || throwable.getCause() instanceof java.net.SocketException
+          || throwable.getCause() instanceof java.net.SocketTimeoutException) {
+        return true;
+      }
+      throwable = throwable.getCause();
+    }
+    return false;
+  }
+
   private void sleepForAttempt(int attempt) {
+    long delay = 1000;
+    long maxDelay = 120_000;
+    // exponential backoff
+    for (int i=0; i<attempt; i++) {
+      delay *= 2;
```

Comment:
2 is aggressive, yes, but this code is handling an uncommon scenario: we've been throttled to the point that we're not even getting a retryable 503, but something cut the connection entirely. It feels appropriate to have an uncommon backoff amount.

I changed the delay computation to match your latest suggestion. I like it best because this way we don't need to add more state to the CloudStorageReadChannel.